Feature/osu 1385 aave

skimaharvey opened the pull request:
What does this PR introduce? 

- [x] AaveV3Strategy 
- [X] AaveV3StrategyTest
- [x] AaveV3StrategyFactory 
- [x] AaveV3StrategyFactoryTest

skimaharvey committed:
feat: create AaveV3CompounderStrategy

skimaharvey committed:
test: create AaveV3DonatingStrategyTest

skimaharvey committed:
feat: create AaveV3StrategyFactory

skimaharvey committed:
test: create AaveV3StrategyFactoryTest

skimaharvey committed:
refactor: remove unused interfaces

skimaharvey committed:
refactor: add AToken address validation

skimaharvey committed:
refactor: add @notice to natspecs

skimaharvey committed:
test: fix failing tests in AaveV3StrategyFactoryTest

skimaharvey committed:
refactor: fix getReserveCaps returned data

skimaharvey committed:
fix: check pool liquidity in availableWithdrawLimit

skimaharvey committed:
refactor: re-use baseimplementation tests

skimaharvey committed:
refactor: add warning comments to AAVE strategy

0xferit reviewed (dismissed):
LGTM, approving. 👍

Note: I’m planning to push a follow‑up commit that adds a constructor check to ensure the passed aToken matches the reserve from Aave’s data provider (to avoid misconfiguration). You can decide whether to include it.

0xferit committed:
fix: derive aToken from pool registry instead of constructor param

- Remove _aToken constructor parameter from AaveV3Strategy
- Add getReserveTokensAddresses to IPoolDataProvider interface
- Derive aToken from dataProvider.getReserveTokensAddresses()
- Add zero-address validation for _addressesProvider
- Remove AUSDC_V3 constant from factory
- Update pragma to >=0.8.25 for consistency
- Update test to reflect new constructor signature

This eliminates the possibility of passing a mismatched aToken
that could cause incorrect balance reporting.

qGolem commented on src/factories/AaveV3StrategyFactory.sol:
```diff
@@ -0,0 +1,98 @@
+// SPDX-License-Identifier: AGPL-3.0-or-later
+pragma solidity >=0.8.25;
+
+import { Create2 } from "@openzeppelin/contracts/utils/Create2.sol";
```
looks like unsused import?

qGolem reviewed (approved):
LGTM, nit for unused import
